src/averageAges.js

```diff
@@ -37,7 +51,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let filtredWomen = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    filtredWomen = filtredWomen
+      .filter(women => people.some(person => person.mother === women.name));
+  }
+
+  const calculateTotalAge = filtredWomen
+    .reduce((sum, person) => sum + (person.died - person.born), 0);
+
+  const calculateAverageAge = calculateTotalAge / filtredWomen.length;
+
+  return calculateAverageAge;
```

Comment:
You already have logic for calculating average age for the list of people (in `calculateMenAverageAge`)
Consider creating a helper function for this (DRY principle)

Reply:
Thanks, very usefull comment